### ISSUE TITLE
DeriNet._roots is not updated

### ISSUE THREAD

**Issue by vidraj** [vidraj/derinet#6]
During database load, the field DeriNet._roots is initialized to contain a list of internal IDs of all cluster roots. But as far as I can see, it is not updated when adding derivations, so the information goes out of date quickly as clusters are merged or split. The information is used when saving the database, possibly resulting in a broken file.

I didn't actually test any of this, I just noticed it reading the source. If I'm wrong, correct me, please.

**Comment by vidraj** [comment]
Commit 29743381a47cd2ec218adc67fdf3053a1161579 possibly fixes a part of this issue, that the roots weren't correctly created on DB load and when adding lexemes. An internal ID is now added to ._roots when the new node doesn't have a parent, i.e. is a root.

I wrote tests for both this and the not-updated-when-adding-derivations issue.

**Comment by vojtsek** [comment]
I followed with commit e3e89bb5ba7410032e1d9aaf89cfb5809257ca0d. It changes _roots representation to set to achieve efficient removal. Changes in the roots that are caused by derivations should be hnadled now. I leave this issue open for now till it will be confirmed.

**Comment by vidraj** [comment]
The roots should probably be updated in sort() as well.

**Comment by vojtsek** [comment]
You are right @vidraj, thanks. It should be easy to add, I will also create a test for that.

**Comment by vidraj** [comment]
Thank you for solving this. :-)
